src/main/java/games/strategy/engine/framework/startup/login/ClientLogin.java

```diff
@@ -17,6 +28,7 @@
   public static final String ENGINE_VERSION_PROPERTY = "Engine.Version";
   public static final String JDK_VERSION_PROPERTY = "JDK.Version";
   public static final String PASSWORD_PROPERTY = "Password";
+  public static final String ENCRYPTED_PASSWORD_PROPERTY = "RSA Encrypted Password";
```

Comment:
And using an HMAC is probably better than the vanilla encryption I suggested above, where the nonce now becomes the message to be authenticated.  So it might require sending a salt in the challenge, as well, to be used for deriving the `SecretKey` from the password (e.g. using `"HmacSHA512"` for the `Mac` and `"PBKDF2WithHmacSHA512"` for the `SecretKeyFactory`).